### PR TITLE
MAUI-1445: Prepare UG - Appearance.

### PR DESCRIPTION
**Pull request by RasikaPalanisamy**
Added the viewappearance.md file for MAUI ListView.

### CODE REVIEW

File: MAUI/ListView/viewappearance.md [hunk 1]
```diff
@@ -0,0 +1,1208 @@
...
+title: Appearance in .NET MAUI ListView control | Syncfusion
+description: Learn here all about Appearance support in Syncfusion .NET MAUI ListView (SfListView) control and more.
+platform: MAUI
+control: SfListView
+documentation: ug
+---
+
+# Appearance in .NET MAUI ListView (SfListView)
+
+The `SfListView` allows customizing appearance of the underlying data, and provides different functionalities to the end-user.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:12]: The `SfListView` allows customizing the appearance of the underlying data, and provides different functionalities to the end-user.

File: MAUI/ListView/viewappearance.md [hunk 2]
```diff
@@ -0,0 +1,1208 @@
...
+
+## Item template
+
+A template can be used to present the data in a way that makes sense for the application by using different controls. `SfListView` allows customizing appearance of view by setting the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) property.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:16]: A template can be used to present the data in a way that makes sense for the application by using different controls. `SfListView` allows customizing the appearance of the view by setting the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) property.

File: MAUI/ListView/viewappearance.md [hunk 3]
```diff
@@ -0,0 +1,1208 @@
...
+
+## Data template selector
+
+The `SfListView` allows customizing appearance of each item with different templates based on specific constraints by using the `DataTemplateSelector`. You can choose a `DataTemplate` for each item at runtime based on the value of data-bound property using `DataTemplateSelector`.
+
+Here, an [ItemsCacheLimit](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewLayout.html#Syncfusion_Maui_ListView_ListViewLayout_ItemsCacheLimit) property maintains number of items reusing in the view. This cache limit is used to create and reuse the [ListViewItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewItem.html) if different templates are used in `DataTemplateSelector` for better scrolling performance. Based on this value, `SfListView` creates number of `ListViewItem` for different templates in the view if new template is created while scrolling, and reuses it if same template is used for improving the scrolling performance.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:22]: Here, an [ItemsCacheLimit](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewLayout.html#Syncfusion_Maui_ListView_ListViewLayout_ItemsCacheLimit) property is used to maintain the number of items that can be reused in the view. This cache limit is used to create and reuse the [ListViewItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewItem.html), if different templates are used in `DataTemplateSelector` for better scrolling performance. Based on this value, `SfListView` creates a number of `ListViewItem` for different templates in the view, if a new template is created while scrolling, and reuses them, if the same template is used to improve the scrolling performance.

File: MAUI/ListView/viewappearance.md [hunk 4]
```diff
@@ -0,0 +1,1208 @@
...
+
+The `SfListView` allows customizing appearance of the underlying data, and provides different functionalities to the end-user.
+
+## Item template
+
+A template can be used to present the data in a way that makes sense for the application by using different controls. `SfListView` allows customizing appearance of view by setting the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) property.
+
+## Data template selector
+
+The `SfListView` allows customizing appearance of each item with different templates based on specific constraints by using the `DataTemplateSelector`. You can choose a `DataTemplate` for each item at runtime based on the value of data-bound property using `DataTemplateSelector`.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:20]: The `SfListView` allows customizing the appearance of each item with different templates based on specific constraints by using the `DataTemplateSelector`.

File: MAUI/ListView/viewappearance.md [hunk 5]
```diff
@@ -0,0 +1,1208 @@
...
+
+Here, an [ItemsCacheLimit](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewLayout.html#Syncfusion_Maui_ListView_ListViewLayout_ItemsCacheLimit) property maintains number of items reusing in the view. This cache limit is used to create and reuse the [ListViewItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewItem.html) if different templates are used in `DataTemplateSelector` for better scrolling performance. Based on this value, `SfListView` creates number of `ListViewItem` for different templates in the view if new template is created while scrolling, and reuses it if same template is used for improving the scrolling performance.
+
+### Create a data template selector
+
+Create custom class that inherits from `DataTemplateSelector`, and override the `OnSelectTemplate` method to return the `DataTemplate` for that item. At runtime, the SfListView invokes the `OnSelectTemplate` method for each item and passes the data object as parameter.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:26]: Create a custom class that inherits from `DataTemplateSelector`, and override the `OnSelectTemplate` method to return the `DataTemplate` for that item. At runtime, the SfListView invokes the `OnSelectTemplate` method for each item and passes the data object as a parameter.

File: MAUI/ListView/viewappearance.md [hunk 6]
```diff
@@ -0,0 +1,1208 @@
...
+   }
+
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+### Applying the data template selector
+
+Assign custom `DataTemplateSelector` to the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) of the `SfListView` either in XAML or C#.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:58]: Assign a custom `DataTemplateSelector` to the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) of the `SfListView` in XAML or C#.

File: MAUI/ListView/viewappearance.md [hunk 7]
```diff
@@ -0,0 +1,1208 @@
...
+    {
+        GridView.BindingContext = e.ItemData;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Horizontal list inside vertical list
+
+The `SfListView` allows you to layout the items in horizontal list inside the vertical list. You can load the nested `SfListView` by customizing the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) of outer `SfListView`.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:194]: The `SfListView` allows you to layout the items in a horizontal list within a vertical list. You can load the nested `SfListView` by customizing the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) of outer `SfListView`.

File: MAUI/ListView/viewappearance.md [hunk 8]
```diff
@@ -0,0 +1,1208 @@
...
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Horizontal list inside vertical list
+
+The `SfListView` allows you to layout the items in horizontal list inside the vertical list. You can load the nested `SfListView` by customizing the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) of outer `SfListView`. 
+
+You should define the size for each inner `SfListView` or set the [AutoFitMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_AutoFitModeProperty) of inner SfListView as `Height`, and define the [ItemSize](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemSize) for outer `SfListView`.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:196]: You should either define the size for each inner `SfListView` or set the [AutoFitMode](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_AutoFitModeProperty) of the inner SfListView to `Height`, and define the [ItemSize](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemSize) for outer `SfListView`.

File: MAUI/ListView/viewappearance.md [hunk 9]
```diff
@@ -0,0 +1,1208 @@
...
+<syncfusion:SfListView x:Name="listView" ItemSpacing="5,0,0,0" />
+{% endhighlight %}
+{% highlight c# %}
+listView.ItemSpacing = new Thickness(5, 0, 0, 0)
+{% endhighlight %}
+{% endtabs %}
+
+## Alternate row styling
+
+The `SfListView` allows applying alternate row styling for items by finding the index of the underlying object using `IValueConverter`.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:292]: The `SfListView` allows applying alternate row styling for items by using the `IValueConverter` to find the index of the underlying object.

File: MAUI/ListView/viewappearance.md [hunk 10]
```diff
@@ -0,0 +1,1208 @@
...
+
+    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture)
+    {
+    }
+}
+{% endhighlight %}
+
+## Rounded corner on items
+
+The `SfListView` allows customizing the item appearance like rounded corner by using the `Frame` layout in the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) property. By defining the `CornerRadius` property of frame layout, you can perform the rounded corner for items.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:364]: The `SfListView` allows customizing the item appearance, such as rounded corners, by using the `Frame` layout in the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) property. By defining the `CornerRadius` property of frame layout, you can perform rounded corner for items.

File: MAUI/ListView/viewappearance.md [hunk 11]
```diff
@@ -0,0 +1,1208 @@
...
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Drop shadow effect on items
+
+The `SfListView` allows customizing the item appearance like shadow effect for items by setting the shadow property of frame as true in [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) property.
+
+N> Define the frame within any view inside `ItemTemplate` with around some margin.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:423]: Define the frame within any view inside `ItemTemplate` with some margin around it.

File: MAUI/ListView/viewappearance.md [hunk 12]
```diff
@@ -0,0 +1,1208 @@
...
+            return grid;
+        });
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## ListViewItem customization
+
+The `SfListView` allows customizing the [ListViewItem](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewItem.html) based on the [ItemType](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ItemType.html). To customize the Header, Footer, GroupHeader, LoadMore, and ListViewItem follow the code example.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:488]: Follow the code example to customize the Header, Footer, GroupHeader, LoadMore, and ListViewItem.

File: MAUI/ListView/viewappearance.md [hunk 13]
```diff
@@ -0,0 +1,1208 @@
...
+    protected override void OnDetachingFrom(ContentPage bindable)
+    {
+        listview.ItemTapped -= ListView_ItemTapped;           
+    }
+}
+
+{% endhighlight %}
+{% endtabs %}
+
+The `IsVisible` model property which is bound to the second template will be enabled when tapping the item and disabled when tapping again the same item.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:798]: The 'IsVisible' model property, which is bound to the second template, will be enabled when the item is tapped and disabled when the item is tapped again.

File: MAUI/ListView/viewappearance.md [hunk 14]
```diff
@@ -0,0 +1,1208 @@
...
+
+## show busy indicator on list view
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:800]: Show busy indicator on list view

File: MAUI/ListView/viewappearance.md [hunk 15]
```diff
@@ -0,0 +1,1208 @@
...
+
+The `SfListView` allows displaying the `SfBusyIndicator` when loading the bounded items. The busy indicator can be enabled and disabled by using `IsBusy` property.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:802]: The `SfListView` allows displaying the `SfBusyIndicator` when loading the bounded items. The busy indicator can be enabled and disabled by using the `IsBusy` property.

File: MAUI/ListView/viewappearance.md [hunk 16]
```diff
@@ -0,0 +1,1208 @@
...
+        }
+        IsLoading = false;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+N> When both `SfBusyIndicator` and `SfListView` loaded with same row and column, you need to set `InputTransparent` as `True` to `SfBusyIndicator` in order to pass touch interaction to `SfListView` in iOS platform.
+
+## show busy indicator on list view items
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:857]: Show busy indicator on list view items

File: MAUI/ListView/viewappearance.md [hunk 17]
```diff
@@ -0,0 +1,1208 @@
...
+
+The `SfListView` allows displaying an activity indicator for an item when its data is being loaded in the background. To perform this, load both `ActivityIndicator` and a `Button` in the same row of a `Grid` element inside the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) of the `SfListView`. The busy indicator and button can be enabled and disabled by using properties `IsButtonVisible` and `IsIndicatorVisible` respectively in the model class.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:859]: The `SfListView` allows displaying an activity indicator for an item when its data is being loaded in the background. To perform this, load both `ActivityIndicator` and a `Button` in the same row of a `Grid` element inside the [ItemTemplate](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.SfListView.html#Syncfusion_Maui_ListView_SfListView_ItemTemplate) of the `SfListView`. The busy indicator and button can be enabled and disabled by using the `IsButtonVisible` and `IsIndicatorVisible` properties respectively in the model class.

File: MAUI/ListView/viewappearance.md [hunk 18]
```diff
@@ -0,0 +1,1208 @@
...
+    public void OnPropertyChanged(string name)
+    {
+        if (this.PropertyChanged != null)
+            this.PropertyChanged(this, new PropertyChangedEventArgs(name));
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+Disable the visibility of Description and `ActivityIndicator` initially while adding items into collection.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:936]: initially, disable the visibility of description and `ActivityIndicator` when adding items to a collection.

File: MAUI/ListView/viewappearance.md [hunk 19]
```diff
@@ -0,0 +1,1208 @@
...
+                </Grid>
+                </Frame>
+            </DataTemplate>
+        </sync:SfListView.ItemTemplate>
+    </sync:SfListView>
+</ContentPage>
+{% endhighlight %}
+{% endtabs %}
+
+In the `Clicked` event of the Button, get the row data from its `BindingContext` and alter the bool values accordingly.
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:1004]: In the `Clicked` event of the button, get the row data from its `BindingContext` and alter the bool values accordingly.

File: MAUI/ListView/viewappearance.md [hunk 20]
```diff
@@ -0,0 +1,1208 @@
...
+        model.IsIndicatorVisible = false;
+    }
+  }
+}
+{% endhighlight %}
+{% endtabs %}
+
+## Item animation on appearing
+
+The `SfListView` supports animating the items by using an [OnItemAppearing](https://help.syncfusion.com/cr/maui/Syncfusion.Maui.ListView.ListViewItem.html#Syncfusion_Maui_ListView_ListViewItem_OnItemAppearing) virtual method. It is raised when the items appearing in the view on scrolling, loading, and navigating from one page to another page. To apply the animation effect for items, follow the steps:
```
Comment (laksmipriyamurugesan) [inline MAUI/ListView/viewappearance.md:1099]: To apply the animation effect for items, follow these steps:

File: MAUI/ListView/viewappearance.md (hunk 1 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 2 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 3 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 4 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 5 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 6 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 7 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 8 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 9 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 10 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 11 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 13 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 14 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 15 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md [hunk 21]
```diff
@@ -0,0 +1,1208 @@
...
+            var contact = new Model(CustomerNames[i], ContactNumber[i]);
+            ContactInfo.Add(contact);
+        }
+        IsLoading = false;
+    }
+}
+{% endhighlight %}
+{% endtabs %}
+
+N> When both `SfBusyIndicator` and `SfListView` loaded with same row and column, you need to set `InputTransparent` as `True` to `SfBusyIndicator` in order to pass touch interaction to `SfListView` in iOS platform.
```
[the remark this answers is not in the document]
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 16 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 17 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 18 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 19 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

File: MAUI/ListView/viewappearance.md (hunk 20 above)
Comment (RasikaPalanisamy) [reply MAUI/ListView/viewappearance.md]: I have modified the line as per the concerns provided.

**Comment by RasikaPalanisamy** [comment]
rebuild

**Pull request merged**